🛠️ Repo: Replace strip-ansi with `util.stripVTControlCharacters`

### Feature Request Checklist

- [x] I have read and agree to Mocha's [Code of Conduct](https://github.com/mochajs/mocha/blob/main/.github/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/mochajs/mocha/blob/main/.github/CONTRIBUTING.md)
- [x] I have searched for [related issues](https://github.com/mochajs/mocha/issues?q=is%3Aissue) and [issues with the `faq` label](https://github.com/mochajs/mocha/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3Afaq%20), but none matched my issue.
- [x] I want to provide a PR to resolve this

### Overview

I ran [`eslint-plugin-depend`](https://github.com/es-tooling/eslint-plugin-depend) on the Mocha repo. Here's one of the outputs:

```plaintext
/Users/josh/repos/mocha/docs/_data/usage.js
  3:19  error  "strip-ansi" should be replaced with an alternative package. Read more here: https://github.com/es-tooling/module-replacements/blob/main/docs/modules/strip-ansi.md  depend/ban-dependencies
```

Perhttps://github.com/es-tooling/module-replacements/blob/main/docs/modules/strip-ansi.md:

> [`strip-ansi`](https://www.npmjs.com/package/strip-ansi) is a simple utility function that can be replaced with native APIs.
>
> ...
>
> Added in v16.11.0, `util.stripVTControlCharacters` can be used to strip ANSI escape codes from a string.

### Suggested Solution

One less dependency is always nice IMO! Now v11 is shipped and we don't support Node <18.18, I think we can rely on `node:util` existing.

Proposal: let's drop the `strip-ansi` dependency in favor of `util.stripVTControlCharacters`?

### Alternatives

We could stay on `strip-ansi`, but I see no reason to.

### Additional Info

💖

--- a/docs/_data/usage.js
+++ b/docs/_data/usage.js
@@ -1,6 +1,6 @@
 'use strict';
 
-const stripAnsi = require('strip-ansi');
+const {stripVTControlCharacters} = require('util');
 const {resolve} = require('path');
 const {execSync} = require('child_process');
 
@@ -11,7 +11,7 @@ const flag = '--help';
  * Return the output of `mocha --help` for display
  */
 module.exports = () => {
-  return stripAnsi(
+  return stripVTControlCharacters(
     String(
       execSync(`"${process.execPath}" ${executable} ${flag}`, {
         cwd: resolve(__dirname, '..')

--- a/package.json
+++ b/package.json
@@ -162,7 +162,6 @@
     "rollup-plugin-polyfill-node": "^0.8.0",
     "rollup-plugin-visualizer": "^5.6.0",
     "sinon": "^9.0.3",
-    "strip-ansi": "^6.0.0",
     "unexpected": "^11.14.0",
     "unexpected-eventemitter": "^2.2.0",
     "unexpected-map": "^2.0.0",

